src/Cli/Microsoft.TemplateEngine.Cli/Commands/create/InstantiateCommand.NoMatchHandling.cs

```diff
@@ -152,11 +152,16 @@ private static string InvalidOptionsListToString(IEnumerable<InvalidTemplateOpti
             if (templateGroup.ShortNames.Any())
             {
                 reporter.WriteLine(LocalizableStrings.InvalidParameterTemplateHint);
-                reporter.WriteCommand(
-                    Example
-                        .For<NewCommand>(args.ParseResult)
-                        .WithArgument(NewCommand.ShortNameArgument, templateGroup.ShortNames[0])
-                        .WithHelpOption());
+                var example = Example
+                    .For<NewCommand>(args.ParseResult)
+                    .WithArgument(NewCommand.ShortNameArgument, templateGroup.ShortNames[0])
+                    .WithHelpOption();
+                var language = matchInfos.Where(mi => mi.Language != null).FirstOrDefault()?.Language;
+                if (language != null)
+                {
+                    example.WithOption(language.Option, language.GetValueOrDefault<string>()!);
+                }
+                reporter.WriteCommand(example);
```

Comment:
It would be good to revert the order to be `dotnet new console --language F# -h` here.